### PR TITLE
feat(origin-ui-core): new ui for automatic supply price for devices

### PR DESCRIPTION
**Pull request by m0azam**

### CODE REVIEW

File: packages/origin-backend/src/pods/device/device.controller.ts [hunk 1]
```diff
@@ -110,7 +113,7 @@ export class DeviceController {
 
         if (
             body?.automaticPostForSale === undefined ||
-            (body?.automaticPostForSale &&
+            (body?.automaticPostForSale === undefined &&
```
Comment (m0azam) [inline packages/origin-backend/src/pods/device/device.controller.ts]: add this because variable are boolean

**Comment by josipbagaric** [comment]
@m0azam Looks good to me, but tests need to be fixed.

File: packages/origin-backend-client-mocks/src/DeviceClientMock.ts [hunk 2]
```diff
@@ -77,4 +70,17 @@ export class DeviceClientMock implements IDeviceClient {
...
         this.storage.set(id, device);
     }
+
+    public async getSupplyBy(facilityName: string, status: number): Promise<IDeviceWithRelationsIds[]> {
+        return [...this.storage.values()];
+    }
+
+    public async delete(id: number): Promise<void> {
+        
+    }
```
Comment (josipbagaric) [inline packages/origin-backend-client-mocks/src/DeviceClientMock.ts:80]: There were some `throw()` methods here. What's the reason for removing them and this being empty now?

**Comment by m0azam** [comment]
Screen Shot 2020-05-29 at 10 11 34 PM

File: packages/origin-backend-client-mocks/src/DeviceClientMock.ts (hunk 2 above)
Comment (m0azam) [reply packages/origin-backend-client-mocks/src/DeviceClientMock.ts:80]: do i need put back as though that are reason of test failed

**Review by josipbagaric — APPROVED** [review]

**Pull request merged**